components/mediators/datamapper/org.wso2.carbon.mediator.datamapper.engine/src/main/java/org/wso2/carbon/mediator/datamapper/engine/core/mapper/MappingResource.java

```diff
@@ -108,6 +110,8 @@ private JSFunction createFunction(InputStream mappingConfig, String outputType)
         // execute in engine
         String[] inputRootElementArray = inputRootelement.split(NAMESPACE_DELIMETER);
         String inputRootElement = inputRootElementArray[inputRootElementArray.length - 1];
+        inputRootElement = inputRootElement.replace(':', '_').replace('=', '_').
```

Comment:
shall we make '_' a constant? I suppose 'HYPHEN' can be used for that? That way, if we change the replacement character one day, we'll just need to change it in one place.